pkg/utils/managedresources/builder/managedresources.go

```diff
@@ -199,7 +200,7 @@ func markSecretsAsGarbageCollectable(ctx context.Context, c client.Client, secre
 	return nil
 }
 
-func secretsFromRefs(obj *resourcesv1alpha1.ManagedResource, excludedNames map[string]struct{}) []*corev1.Secret {
+func secretsFromRefs[S sets.Set[string]](obj *resourcesv1alpha1.ManagedResource, excludedNames S) []*corev1.Secret {
 	secrets := make([]*corev1.Secret, 0, len(obj.Spec.SecretRefs))
 	for _, secretRef := range obj.Spec.SecretRefs {
 		if _, ok := excludedNames[secretRef.Name]; !ok {
```

Comment:
You can call `excludedNames.Has(secretRef.Name)` now.

Reply:
I made the suggested changes, although the functionality stays unaffected and one could also argue that this does not improve readability over what the std lib already provides.